plugins/inputs/sql/README.md

```diff
@@ -0,0 +1,143 @@
+
+### Configuration
+
+This section contains the default TOML to configure the plugin.  You can
+generate it using `telegraf --usage <plugin-name>`.
+
+```toml
+[[inputs.sql]]
+  ## Database Driver
+  ## See https://github.com/influxdata/telegraf/blob/master/docs/SQL_DRIVERS_INPUT.md for
+  ## a list of supported drivers.
+  driver = "mysql"
+
+  ## Data source name for connecting
+  ## The syntax and supported options depends on selected driver.
+  dsn = "username:password@mysqlserver:3307/dbname?param=value"
+
+  ## Timeout for any operation
+  # timeout = "5s"
+
+  ## Connection time limits
+  ## By default the maximum idle time and maximum lifetime of a connection is unlimited, i.e. the connections
+  ## will not be closed automatically. If you specify a positive time, the connections will be closed after
+  ## idleing or existing for at least that amount of time, respectively.
+  # connection_max_idle_time = "0s"
+  # connection_max_life_time = "0s"
+
+  ## Connection count limits
+  ## By default the number of open connections is not limited and the number of maximum idle connections
+  ## will be inferred from the number of queries specified. If you specify a positive number for any of the
+  ## two options, connections will be closed when reaching the specified limit. The number of idle connections
+  ## will be clipped to the maximum number of connections limit if any.
+  # connection_max_open = 0
+  # connection_max_idle = auto
+
+  [[inputs.sql.query]]
+    ## Query to perform on the server
+    query="SELECT user,state,latency,score FROM Scoreboard WHERE application > 0"
+    ## Alternatively to specifying the query directly you can select a file here containing the SQL query.
+    ## Only one of 'query' and 'query_script' can be specified!
+    # query_script = "/path/to/sql/script.sql"
+
+    ## Name of the measurement
+    ## In case both measurement and 'measurement_col' are given, the latter takes precedence.
+    # measurement = "sql"
+
+    ## Column name containing the name of the measurement
+    ## If given, this will take precedence over the 'measurement' setting. In case a query result
+    ## does not contain the specified column, we fall-back to the 'measurement' setting.
+    # measurement_col = ""
+
+    ## Column name containing the time of the measurement
+    ## If ommited, the time of the query will be used.
+    # time_col = ""
+
+    ## Format of the time contained in 'time_col'
+    ## The time must be 'unix', 'unix_ms', 'unix_us', 'unix_ns', or a golang time format.
+    ## See https://golang.org/pkg/time/#Time.Format for details.
+    # time_format = "unix"
+
+    ## Column names containing tags
+    ## An empty include list will reject all columns and an empty exclude list will not exclude any column.
+    ## I.e. by default no columns will be returned as tag and the tags are empty.
+    # tag_cols_include = []
+    # tag_cols_exclude = []
+
```

Comment:
can we include something about type information? maybe at the least:

```
int_fields = []
float_fields = []
```

Reply:
We can make them optional at least. For mariadb/mysql I get the column types nicely from the database scheme and if I understand the pgx driver correctly this also hold for PostgreSQL. There seem to be a few though that do not support datatypes correctly, however for those I was thinking of outputting the data as string and use converter plugins... But of course we can also use those options. How about naming them `{float,int,uint,bool,string}_cols` then?

Reply:
yeah, I think we usually name them with the _fields suffix.

Reply:
Should be fixed now. Decided to use `field_columns_{float,int,uint,bool,string}` as convention. Hope that works for you!?